test/e2e/k8s_provisioner_test.go

```diff
@@ -70,5 +84,46 @@ var _ = Describe("k8s provisioner", func() {
 				}).Should(BeTrue())
 			})
 		})
+
+		// TODO(tflannag): add test specs for ensuring the object and metadata ConfigMaps get re-created when deleted
+		It("should re-create underlying system resources", func() {
+			var (
+				pod *corev1.Pod
+			)
+
+			By("getting the underlying bundle unpacking pod")
+			Eventually(func() bool {
+				requirement, err := labels.NewRequirement("core.rukpak.io/owner-name", selection.Equals, []string{bundle.GetName()})
+				if err != nil {
+					return false
+				}
+				pods := &corev1.PodList{}
+				if err := c.List(context.Background(), pods, &client.ListOptions{
+					LabelSelector: labels.NewSelector().Add(*requirement),
+					Namespace:     defaultSystemNamespace,
+				}); err != nil {
+					return false
+				}
+				if len(pods.Items) != 1 {
+					return false
+				}
+				pod = &pods.Items[0]
+				return true
+			}).Should(BeTrue())
+
+			By("storing the pod's original UID")
+			originalUID := pod.GetUID()
+
+			By("deleting the underlying pod and waiting for it to be re-created")
+			Eventually(func() error {
```

Comment:
Nit: do we need `Eventually` here?

Reply:
Ah, just saw your other comment. That's... interesting. Any clue why its flaky without the eventually?

Reply:
I'm not entirely sure - this is what I had logged in a markdown file when I was debugging yesterday:

```
    Expected
        <*errors.StatusError | 0xc0003c0aa0>: {
            ErrStatus: {
                TypeMeta: {Kind: "", APIVersion: ""},
                ListMeta: {
                    SelfLink: "",
                    ResourceVersion: "",
                    Continue: "",
                    RemainingItemCount: nil,
                },
                Status: "Failure",
                Message: "pods \"plain-unpack-bundle-olm-crds-valid\" not found",
                Reason: "NotFound",
                Details: {
                    Name: "plain-unpack-bundle-olm-crds-valid",
                    Group: "",
                    Kind: "pods",
                    UID: "",
                    Causes: nil,
                    RetryAfterSeconds: 0,
                },
                Code: 404,
            },
        }
    to be nil
```

My initial thought was an object cache issue but we create the c-r client.Client using client.New so we're likely directly reading/writing to the apiserver throughout the e2e suite. Maybe the `CreateOrRecreate(...)` function is deleting the Pod when it detects we need to make changes to the underlying bundle unpack pod?

I realize I could also just wrap the Delete call with `client.IgnoreNotFound(c.Delete(ctx, pod))` to burn down the flakes, which seems fine to me, but I'm not sure if there's the possibility we sweep legitimate controller issues under the rug over time.

Reply:
Somewhat mystifying that it exists for the `List` but not the `Get`. If it was a caching issue, `Get` would always have it if a previous `List` did (assuming it didn't get deleted in-between).

`CreateOrRecreate()` is an interesting thought.

> there's the possibility we sweep legitimate controller issues under the rug over time.
I think this is deserving of a little more attention for this reason.

Reply:
Added a TODO comment for re-evaluating this in #105. Any existing TODO comments should be created as issues in #106.

Reply:
I think I figured it out: https://github.com/operator-framework/rukpak/pull/107